### PR TITLE
Make finding aggregator_node runfile bzlmod compatible

### PR DESCRIPTION
**Pull request by lalten**
Extracted from / in preparation for https://github.com/mvukov/rules_ros2/pull/238

No functional or version changes in this one.

### CODE REVIEW

File: ros2/test/diagnostics/tests.py [hunk 1]
```diff
@@ -21,6 +21,10 @@
 import launch_testing.asserts
 import launch_testing.markers
 import rclpy
+from python.runfiles import Runfiles
```
Comment (mvukov) [inline ros2/test/diagnostics/tests.py:24]: Per the google style guide: e.g. `import python.runfiles`; then use `python.runfiles.Runfiles`.

File: ros2/test/diagnostics/tests.py [hunk 2]
```diff
@@ -21,6 +21,10 @@
...
+
+AGGREGATOR_NODE_PATH = Runfiles.Create().Rlocation(
+    'ros2_diagnostics/aggregator_node')
```
Comment (mvukov) [inline ros2/test/diagnostics/tests.py:27]: This is interesting. I am curious what reads the path now. Can you paste the path here, please? Just for the future reference.

Comment (lalten) [reply ros2/test/diagnostics/tests.py:27]:
```
# Workspace
working dir:  /home/laurenz/.cache/bazel/_bazel_laurenz/560f8735b75dc5afa81db7c5a8d429bc/sandbox/linux-sandbox/169/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/ros2/test/diagnostics/diagnostic_heartbeat_test.runfiles/com_github_mvukov_rules_ros2
runfile path: /home/laurenz/.cache/bazel/_bazel_laurenz/560f8735b75dc5afa81db7c5a8d429bc/sandbox/linux-sandbox/196/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/ros2/test/diagnostics/diagnostic_heartbeat_test.runfiles/ros2_diagnostics/aggregator_node

# Bzlmod (on the bzlmod branch)
working dir:  /home/laurenz/.cache/bazel/_bazel_laurenz/560f8735b75dc5afa81db7c5a8d429bc/sandbox/linux-sandbox/166/execroot/_main/bazel-out/k8-fastbuild/bin/ros2/test/diagnostics/diagnostic_heartbeat_test.runfiles/_main
runfile path: /home/laurenz/.cache/bazel/_bazel_laurenz/560f8735b75dc5afa81db7c5a8d429bc/sandbox/linux-sandbox/162/execroot/_main/bazel-out/k8-fastbuild/bin/ros2/test/diagnostics/diagnostic_heartbeat_test.runfiles/_main~non_module_deps~ros2_diagnostics/aggregator_node
```

File: ros2/test/diagnostics/tests.py (hunk 1 above)
Comment (lalten) [reply ros2/test/diagnostics/tests.py]: done

**Review by mvukov — APPROVED** [review]

**Pull request merged**